diff --git a/node_modules/element-plus/es/components/tooltip/src/content2.mjs b/node_modules/element-plus/es/components/tooltip/src/content2.mjs
index b02ab74..9ba9119 100644
--- a/node_modules/element-plus/es/components/tooltip/src/content2.mjs
+++ b/node_modules/element-plus/es/components/tooltip/src/content2.mjs
@@ -64,7 +64,7 @@ const _sfc_main = /* @__PURE__ */ defineComponent({
     const ariaHidden = ref(true);
     const onTransitionLeave = () => {
       onHide();
-      isFocusInsideContent() && tryFocus(document.body);
+      isFocusInsideContent() && tryFocus(document.activeElement.parentElement);
       ariaHidden.value = true;
     };
     const stopWhenControlled = () => {
@@ -112,7 +112,7 @@ const _sfc_main = /* @__PURE__ */ defineComponent({
       var _a;
       const popperContent = (_a = contentRef.value) == null ? void 0 : _a.popperContentRef;
       const activeElement = (event == null ? void 0 : event.relatedTarget) || document.activeElement;
-      return popperContent == null ? void 0 : popperContent.contains(activeElement);
+      return popperContent && ((activeElement.shadowRoot || activeElement).contains(popperContent) || popperContent.contains(activeElement));
     };
     watch(() => unref(open), (val) => {
       if (!val) {
