diff --git a/node_modules/element-plus/es/components/tooltip/src/content.vue2.mjs b/node_modules/element-plus/es/components/tooltip/src/content.vue2.mjs
index 13f1c16..87f31ec 100644
--- a/node_modules/element-plus/es/components/tooltip/src/content.vue2.mjs
+++ b/node_modules/element-plus/es/components/tooltip/src/content.vue2.mjs
@@ -71,7 +71,7 @@ var _sfc_main = /* @__PURE__ */ defineComponent({
     const ariaHidden = ref(true);
     const onTransitionLeave = () => {
       onHide();
-      isFocusInsideContent() && focusElement(document.body, { preventScroll: true });
+      isFocusInsideContent() && focusElement(document.activeElement.parentElement, { preventScroll: true });
       ariaHidden.value = true;
     };
     const stopWhenControlled = () => {
@@ -107,7 +107,19 @@ var _sfc_main = /* @__PURE__ */ defineComponent({
       var _a;
       const popperContent = (_a = contentRef.value) == null ? void 0 : _a.popperContentRef;
       const activeElement = (event == null ? void 0 : event.relatedTarget) || document.activeElement;
-      return popperContent == null ? void 0 : popperContent.contains(activeElement);
+      if (popperContent.dataset?.tikiFocused) {
+        return false;
+      }
+
+      if (popperContent && ((activeElement.shadowRoot || activeElement).contains(popperContent) || popperContent.contains(activeElement))) {
+        popperContent.dataset.tikiFocused = "true";
+        setTimeout(() => {
+          delete popperContent.dataset.tikiFocused;
+        }, 100);
+        return true;
+      }
+
+      return false;
     };
     watch(
       () => unref(open),
