commit 00f4bee051fc7c0b87e907d8d6fb0506c0ca203c
Author: SoftStart Code <code@softstart.io>
Date:   Sun Dec 11 23:39:26 2022 +0000

    [FIX] Warnings for curly brackets access is deprecated

diff --git a/SmartyLint/Cli.php b/SmartyLint/Cli.php
index a72aae5..e691296 100644
--- a/SmartyLint/Cli.php
+++ b/SmartyLint/Cli.php
@@ -70,14 +70,14 @@ class SmartyLint_Cli {
                 continue;
             }
 
-            if ($arg{0} === '-') {
+            if ($arg[0] === '-') {
                 if ($arg === '-' || $arg === '--') {
                     // Empty argument, ignore it.
                     continue;
                 }
 
 
-                if ($arg{1} === '-') {
+                if ($arg[1] === '-') {
                     $values
                         = $this->processLongArgument(substr($arg, 2), $i, $values);
                 } else {
@@ -186,7 +186,7 @@ class SmartyLint_Cli {
      */
     public function processUnknownArgument($arg, $pos, $values) {
         // We don't know about any additional switches; just files.
-        if ($arg{0} === '-') {
+        if ($arg[0] === '-') {
             echo 'ERROR: option "'.$arg.'" not known.'.PHP_EOL.PHP_EOL;
             $this->printUsage();
             exit(2);
diff --git a/SmartyLint/CommentParser/CommentElement.php b/SmartyLint/CommentParser/CommentElement.php
index b9e86c2..00c8800 100644
--- a/SmartyLint/CommentParser/CommentElement.php
+++ b/SmartyLint/CommentParser/CommentElement.php
@@ -134,7 +134,7 @@ class SmartyLint_CommentParser_CommentElement extends SmartyLint_CommentParser_S
         for ($i = $pos; $i < $count; $i++) {
             $content = trim($this->tokens[$i]);
             if ($content !== '') {
-                if ($content{0} === '@') {
+                if ($content[0] === '@') {
                     return -1;
                 }
 
diff --git a/SmartyLint/CommentParser/FileCommentParser.php b/SmartyLint/CommentParser/FileCommentParser.php
index 6116dc0..dac4787 100644
--- a/SmartyLint/CommentParser/FileCommentParser.php
+++ b/SmartyLint/CommentParser/FileCommentParser.php
@@ -158,7 +158,7 @@ class SmartyLint_CommentParser_FileCommentParser {
                     $line = substr($line, $lEnd);
                 } else if (substr($line, -($rEnd), $rEnd) === ('*'.$this->smartylFile->rDelimiter)) {
                     $line = substr($line, 0, -($rEnd));
-                } else if ($line{0} === '*') {
+                } else if ($line[0] === '*') {
                     $line = substr($line, 1);
                 }
 
@@ -197,7 +197,7 @@ class SmartyLint_CommentParser_FileCommentParser {
             if (trim($word) !== '') {
                 $wordWasEmpty = false;
             }
-            if ($word{0} === '@') {
+            if ($word[0] === '@') {
                 $tag = substr($word, 1);
 
                 // Filter out @ tags in the comment description.
